include/boost/url/impl/url_view_base.ipp

```diff
@@ -548,35 +548,29 @@ int
 url_view_base::
 compare(const url_view_base& other) const noexcept
 {
-    int comp = detail::ci_compare(
-        scheme(),
-        other.scheme());
+    int comp = has_scheme() - other.has_scheme();
     if ( comp != 0 )
         return comp;
 
-    comp = detail::compare_encoded(
-        encoded_user(),
-        other.encoded_user());
-    if ( comp != 0 )
-        return comp;
-
-    comp = detail::compare_encoded(
-        encoded_password(),
-        other.encoded_password());
-    if ( comp != 0 )
-        return comp;
+    if (has_scheme())
+    {
+        comp = detail::ci_compare(
```

Comment:
we need this as a public function in `grammar` if it is a case-insensitive comparison

Reply:
Where? What file?